### ISSUE TITLE
mro problem when building Config in complex model inheritance

### ISSUE THREAD

**Issue by Alcolo47** [pydantic/pydantic#2568]
Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.8.1
            pydantic compiled: False
               python version: 3.8.6 | packaged by conda-forge | (default, Oct  7 2020, 19:08:05)  [GCC 7.5.0]
                     platform: Linux-5.10.23-1-MANJARO-x86_64-with-glibc2.10
     optional deps. installed: ['dotenv', 'typing-extensions']
```

```
from pydantic.main import BaseModel

class A(BaseModel): pass
class B(A): pass
class C(BaseModel): pass
class D(B, C): pass
class E(C): pass
class F(B, BaseModel): pass
class Z(D, E, F): pass
```
Gives "TypeError: Cannot create a consistent method resolution order (MRO) for bases Config, Config"

But if you replace "from pydantic.main import BaseModel" with "class BaseModel: pass" all work fine (proof of that this hierarchy are mro compliant)

**Comment by samuelcolvin** [comment]
I think this is the same as #1092, although better explained here (thus I'll leave this issue open).

But since this is quite fundamental to pydantic and doesn't have a direct effect on users, I'd like to wait for v2 to fix this.

**Comment by caniko** [comment]
> I think this is the same as #1092, although better explained here (thus I'll leave this issue open).
> 
> But since this is quite fundamental to pydantic and doesn't have a direct effect on users, I'd like to wait for v2 to fix this.

I am dependent on complex hierarchies in my projects, what do you mean that it doesn't affect the users? Your elaboration is appreciated, considering that I am a user.

**Comment by joshuadavidson** [comment]
I've run up against this problem as well. When I follow the trail for the implementation in v2 I don't see a PR or any code for how exactly it was fixed. Is the fix something that can be patched manually in v1.9? Is there any workaround?

**Comment by samuelcolvin** [comment]
Fixed in V2 as we no longer allow config as a class.

No work around that I know of.
